src/js/components/perfPages/perfTableUtils.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import queryString from 'query-string';
+import is from 'sarcastic';
+
+// This differs from the normal query paramemter helper classes in that
+// it is designed to have default parameters as state. Also, its fairly
+// strongly typed.
+export class QueryParamHelpers {
+  default_params: { [string]: mixed };
+
+  constructor(default_params: { [string]: mixed }) {
+    this.default_params = default_params;
+    this.get = this.get.bind(this); // Javascript grossness!
```

Comment:
Hm, you're using the public class fields syntax (that is, an arrow fn to define `get`), so I don't think this line is needed.

https://reactjs.org/docs/handling-events.html